packages/libs/eda/src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx

```diff
@@ -517,6 +515,38 @@ function ScatterplotViz(props: VisualizationProps<Options>) {
     (vizConfig.valueSpecConfig === 'Smoothed mean with raw' ||
       vizConfig.valueSpecConfig === 'Best fit line with raw');
 
+  // non-zero baseline for continuous overlay variable
+  const defaultOverlayMinMax = useDefaultAxisRange(
+    overlayVariable,
+    undefined,
+    undefined,
+    undefined,
+    vizConfig.dependentAxisLogScale,
+    'Full',
+    // non-zero baseline
+    true
+  );
+
+  // If numeric overlay, record the min and max and make a value to color map function
+  const overlayMin: number | undefined = defaultOverlayMinMax?.min as number;
+  const overlayMax: number | undefined = defaultOverlayMinMax?.max as number;
+  // Diverging colorscale, assume 0 is midpoint. Colorscale must be symmetric around the midpoint
+  const maxAbsOverlay =
+    Math.abs(overlayMin) > overlayMax ? Math.abs(overlayMin) : overlayMax;
+  const gradientColorscaleType:
+    | 'sequential'
+    | 'sequential reversed'
+    | 'divergent'
+    | undefined =
+    overlayMin != null && overlayMax != null
+      ? overlayMin >= 0 && overlayMax >= 0
+        ? 'sequential'
+        : overlayMin <= 0 && overlayMax <= 0
+        ? 'sequential reversed'
+        : 'divergent'
+      : undefined;
+  let overlayValueToColorMapper: ((a: number) => string) | undefined;
```

Comment:
This should go inside the `data = usePromise(...)` for two reasons
1. this particular variable is set inside the usePromise (a side-effect that we don't want) but is not accessed later...
2. ...because the overlayValueToColorMapper is returned as part of the `data` payload (from whence it is used to make the legend)